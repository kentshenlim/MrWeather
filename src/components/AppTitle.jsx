import PropTypes from 'prop-types';
import { GitHub } from 'react-feather';
import { styled } from 'styled-components';

import mrWeather from '../assets/img/mrWeather.png';
import color from '../styles/color';
import fontSize from '../styles/fontSize';

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 12rem;
  > h1 {
    margin: 0;
    font-size: ${fontSize.normal};
  }
  & a {
    color: ${color.accent};
  }
  & img {
    width: 80%;
    transition: all 1s ease-in-out;
  }
  & .rotate {
    animation-name: rotate;
    animation-duration: 2s;
    animation-iteration-count: infinite;
  }
  & .inverted {
    transform: rotate(180deg);
  }
  @keyframes rotate {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
  @media only screen and (max-width: 520px) {
    width: 7rem;
    > h1 {
      font-size: ${fontSize.small};
    }
    > h1 > :first-child {
      display: none;
    }
  }
`;

export default function AppTitle({ locationStatus }) {
  return (
    <Wrapper>
      <h1>
        <GitHub />
        <a
          href="https://github.com/kentshenlim/MrWeather"
          target="_blank"
          rel="noreferrer"
        >
          &nbsp;MrWeather
        </a>
      </h1>
      <img
        src={mrWeather}
        alt="MrWeather Icon"
        draggable={false}
        className={
          locationStatus == 'loading'
            ? 'rotate'
            : locationStatus == 'error'
            ? 'inverted'
            : ''
        }
      />
    </Wrapper>
  );
}

AppTitle.propTypes = {
  locationStatus: PropTypes.string.isRequired,
};
