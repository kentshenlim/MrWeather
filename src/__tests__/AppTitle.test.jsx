import { it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import AppTitle from '../components/AppTitle';

it('Title is MrWeather, icon image must be present', () => {
  render(<AppTitle />);

  expect(
    screen.getByRole('heading', { name: /mrweather/i })
  ).toBeInTheDocument();
  expect(
    screen.getByRole('img', { name: /mrweather icon/i })
  ).toBeInTheDocument();
});

it('Img should be bounded by anchor tag, href is GitHub repo link', () => {
  render(<AppTitle />);

  const anchor = screen.getByRole('link', { name: /mrweather icon/i });
  expect(anchor).toBeInTheDocument();
  expect(anchor).toContainElement(
    screen.getByRole('img', { name: /mrweather icon/i })
  );
  expect(anchor).toHaveAttribute(
    'href',
    expect.stringContaining('https://github.com')
  );
  expect(anchor).toHaveAttribute('target', '_blank');
});

it('Match snapshot', () => {
  const { container } = render(<AppTitle />);

  expect(container).toMatchSnapshot();
});
