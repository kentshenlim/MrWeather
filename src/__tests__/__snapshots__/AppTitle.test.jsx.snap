// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Match snapshot 1`] = `
<div>
  <div
    class="sc-bcPKhP hNWpLB"
  >
    <h1>
      MrWeather
    </h1>
    <a
      href="https://github.com/kentshenlim/MrWeather"
      target="_blank"
    >
      <img
        alt="MrWeather Icon"
        src="/src/assets/img/mrWeather.png"
      />
    </a>
  </div>
</div>
`;

exports[`Snapshot test > Match snapshot 1`] = `
<div>
  <div
    class="sc-bcPKhP hNWpLB"
  >
    <h1>
      MrWeather
    </h1>
    <a
      href="https://github.com/kentshenlim/MrWeather"
      target="_blank"
    >
      <img
        alt="MrWeather Icon"
        src="/src/assets/img/mrWeather.png"
      />
    </a>
  </div>
</div>
`;
